pkg/internal/common/logging/modlog/provider.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package modlog
+
+import (
+	"fmt"
+	"log"
+	"os"
+
+	logapi "github.com/hyperledger/aries-framework-go/pkg/common/log"
+)
+
+//providerOpts contains options for initializing Moduled logging provider
+type providerOpts struct {
+	customProvider logapi.LoggerProvider
+}
+
+//ModuledLoggerOpts is option to provide logger provider opts for initializing Moduled logging provider
+type ModuledLoggerOpts func(opts *providerOpts)
+
+//WithCustomProvider can be used to provide custom logger provider for moduled logger
+func WithCustomProvider(customProvider logapi.LoggerProvider) ModuledLoggerOpts {
+	return func(opts *providerOpts) {
+		opts.customProvider = customProvider
+	}
+}
+
+//ModuledLoggerProvider returns logger provider for moduled level based logger
```

Comment:
Isn’t this NewProvider? Also why return interface instead of *provider?

Reply:
spilt it into --> NewModLogProvider & NewDefLogProvider

Reply:
Why not return *provider?